woocommerce/woocommerce-ios
# [Mobile Payments] Add ChargeID to Order

Part of: #5976 


### PR Length
Please don't be too worried about the length of this PR - the majority of the changes are test data (+183) or changing to use `Order.fake()` instead of `Order()` when building test data (+207, -416). This will mean that future changes to `Order` don't require large numbers of changes in other test files.

When changing to use `Order.fake().copy()` to change the value of the sample data, I have removed the unnecessary parameters from the call which would have explicitly set a property to the default empty value that `.fake()` sets in the first place.

### Description


We need the `chargeID` to be able to fetch `/payments/charges/<charge_id>`, which will give us various details about a payment that's been taken, in particular the last 4 digits of the card, and the card type.

This adds the `chargeID` to `Networking.Order`, and adds `meta_data` to the `_fields` when loading a single order so that we receive the `charge_id`.

Further PRs will add the ability to fetch a `Charge`, which we will use when displaying payment details in the UI. This will also be beneficial for fetching details for printing receipts, avoiding the need to store that locally on device and the limitation that IPP receipts must be printed from the same device as the payment was taken from.

### Testing instructions

Consider only testing as a part of the further changes where we will fetch the details of charges for showing the last 4 digits of a card number in the order details.

You can test the decoding by running the app from Xcode, with a breakpoint in [`Order.init(from:)`](https://github.com/woocommerce/woocommerce-ios/blob/a4c10b8982207f0542f5e1477d451b58b968a5b1/Networking/Networking/Model/Order.swift#L148) after the `chargeID` is decoded.

1. Enable the simulated card reader
2. Launch the app and go to `Orders`
3. Tap + and take an IPP for the order you add
4. Return to the `Orders` root screen, and enable your breakpoint
5. Tap the order, and you should hit the breakpoint and be able to see that a `chargeID` is decoded. Typically these look like: `ch_3KMtak2EdyGr1FMV02G9Qqq1` 

### Screenshots

N/A no UI changes

---
